### PR TITLE
Fixed Config API to handle PhyIoAdapters better - added systemAdapters as a map inside PhysicalIo

### PR DESCRIPTION
**Pull request by kalyan-nidumolu**
Fixed Config API to handle PhyIoAdapters better:

1) Added systemAdapters as a map inside PhysicalIo
   - Instead of having a separate array of system adapters, we are
     moving this to be part of PhysicalIo. ie., each PhysicalIo
     will now also have a map of system adapters using the
     PhysicalIoAdapter.

2) Added some comments to other fields in devmodel.

### CODE REVIEW

**Comment by rvs** [comment]
I see you only changed the proto file @kalyan-nidumolu -- don't you also plan to change the go/python under api?

**Comment by kalyan-nidumolu** [comment]
I thought I had to do it in different PRs?

Thanks,
Kalyan

**Comment by kalyan-nidumolu** [comment]
Pushed the generated .go files also..

Thanks,
Kalyan

File: api/proto/config/devmodel.proto [hunk 1]
```diff
@@ -47,19 +47,19 @@ message sWAdapterParams {
...
 // can be shared between workload/services running on the node.
 message SystemAdapter {
         // name of the adapter; hardware-specific e.g., eth0
         string name = 1;
 
         // deprecated; need level 2 spec. sWAdapterParams allocDetails = 20;
-
         // this is part of the freelink group
-        // deprecate: have a separate device policy object in the API
+        // DEPRECATED by PhyIoAdapter.usagePolicy.
```
Comment (eriknordmark) [inline api/proto/config/devmodel.proto]: Can you comment out the definition as well?

File: api/proto/config/devmodel.proto [hunk 2]
```diff
@@ -47,19 +47,19 @@ message sWAdapterParams {
...
         bool freeUplink = 2;
 
+        // uplink - DEPRECATED by PhysicalIO.Usage / PhysicalIO.UsagePolicy
         // this is part of the uplink group
         // deprecate: have a separate device policy object in the API
         bool uplink = 3;
```
Comment (eriknordmark) [inline api/proto/config/devmodel.proto]: Ditto - does anything in EVE refer to this field? If not, how about commenting it out?

File: api/proto/config/devmodel.proto [hunk 3]
```diff
@@ -68,11 +68,8 @@ message SystemAdapter {
         string networkUUID = 4;
 
         // if its static network we need ip address
+        // If this is specified, networkUUID must be ""
```
Comment (eriknordmark) [inline api/proto/config/devmodel.proto]: I thought I commented on this already, but I think the comment is reversed. networkUUID must be set to a valid UUID if the IP address is set.

File: api/proto/config/devmodel.proto [hunk 4]
```diff
@@ -68,11 +68,8 @@ message SystemAdapter {
         string networkUUID = 4;
```
Comment (eriknordmark) [inline api/proto/config/devmodel.proto]: Would make sense to add a comment that if this is not set, then the device will treat the port as an app-shared L2 port i.e., the device will not configure any IP on the port.

File: api/proto/config/devmodel.proto [hunk 5]
```diff
@@ -141,12 +138,21 @@ message PhysicalIO {
...
+      //    app traffic?? etc..
       PhyIOUsagePolicy usagePolicy    = 7;
 
       // physical and logical attributes
       //    For example in WWAN to which firmware version to laod etc
       map <string, string> cbattr = 8;
+
+      // systemAdapters - List of system Adapters using this IoAdapter.
+      // key: SystemAdapter.Name
+      map <string, SystemAdapter> systemAdapters = 11;
```
Comment (eriknordmark) [inline api/proto/config/devmodel.proto]: Why is this inside each physicalIO as opposed to being a global map?

File: api/proto/config/devmodel.proto (hunk 1 above)
Comment (kalyan-nidumolu) [reply api/proto/config/devmodel.proto]: Looks like no one is using this anyways.. Will remove this completely and just keep the number reserved in the comment.

Comment (eriknordmark) [reply api/proto/config/devmodel.proto]: Better to see all of the changes. 
Sometimes I've deferred pushing the separate commit of the generated files to make life easier for the code reviewers, but there are no technical constraints to do all of it at once.

File: api/proto/config/devmodel.proto (hunk 3 above)
Comment (eriknordmark) [reply api/proto/config/devmodel.proto]: In general that is not required, for instance if we ever support pt-pt interfaces.
So we don't need to add that check on the device.

Comment (eriknordmark) [reply api/proto/config/devmodel.proto]: But note that my comment is to just change from 'must be ""'  to 'must be set'

File: api/proto/config/devmodel.proto (hunk 1 above)
Comment (kalyan-nidumolu) [reply api/proto/config/devmodel.proto]: Ok Cool.. I committed all the changes.

freeUplink -- I need to make a few changes.. will do it in a separate commit.. along with the support for the new policy fields..

File: api/proto/config/devmodel.proto (hunk 3 above)
Comment (kalyan-nidumolu) [reply api/proto/config/devmodel.proto]: Changed it as:

        // addr - if its static network we need ip address
        // If this is specified, networkUUID must also be specified. addr
        // is expected to be in sync with the network object ( same subnet etc ).

File: api/proto/config/devmodel.proto (hunk 5 above)
Comment (kalyan-nidumolu) [reply api/proto/config/devmodel.proto]: Removed this.. made the changes as per our discussion.

**Pull request merged**